database_cleanup/tests/test_database_cleanup.py

```diff
@@ -78,3 +89,9 @@ def test_database_cleanup(self):
         with self.assertRaises(ProgrammingError):
             with self.registry.cursor() as cr:
                 self.env.cr.execute('select * from database_cleanup_test')
+        with self.registry.cursor() as cr2:
+            # Release the delete of ir_module_module pending
+            self.env.cr.rollback()
+            cr2.execute(
+                "DELETE FROM ir_module_module WHERE id=%s", (module.id,))
```

Comment:
if this is necessary, the test has failed. Or better: Then we have the problem that the commits the module needs to do shouldn't be committed. Maybe it's better to defuse `cr.commit` at the beginning of the test and reinstate it here?